scripts/google_sheet_uploader/uploader.py

```diff
@@ -0,0 +1,156 @@
+"""Module containing the GoogleSheetUploader class for uploading CSV data to Google Sheets."""
+
+import os
+import pandas as pd
+from googleapiclient.discovery import build
+from google.oauth2 import service_account
+import logging
+from typing import Any
+
+
+class GoogleSheetUploaderError(Exception):
+    """Base exception class for GoogleSheetUploader."""
+
+    pass
+
+
+class CSVFileNotFoundError(GoogleSheetUploaderError):
+    """Exception raised when a CSV file is not found."""
+
+    pass
+
+
+class CSVReadError(GoogleSheetUploaderError):
+    """Exception raised when a CSV file cannot be read."""
+
+    pass
+
+
+class SheetClearError(GoogleSheetUploaderError):
+    """Exception raised when clearing a sheet fails."""
+
+    pass
+
+
+class SheetWriteError(GoogleSheetUploaderError):
+    """Exception raised when writing to a sheet fails."""
+
+    pass
+
+
+class GoogleSheetUploader:
+    """Uploads CSV files to specified Google Sheets tabs."""
+
+    def __init__(self, config: Any):
+        """Initialize the uploader with configuration and set up Google Sheets service.
+
+        Args:
+            config (Config): Configuration object containing mappings and credentials.
+        """
+        self.config = config
+        self.logger = logging.getLogger(__name__)
+
+        try:
+            self.credentials = service_account.Credentials.from_service_account_file(
+                self.config.service_account_file,
+                scopes=["https://www.googleapis.com/auth/spreadsheets"],
+            )  # type: ignore
+            self.service = build("sheets", "v4", credentials=self.credentials)
+        except Exception as e:
```

Comment:
nit: Do you happen to know which errors exactly we should catch? Catchalls aren't generally a good idea.